### PR TITLE
FIX: add guardrails for too big tk figures

### LINKED ISSUE #17460: set_size_inches with a width over 14090 crashes Xorg
### Bug report

**Bug summary**

Calling set_size_inches with w=14090.5 causes Xorg to lock up and corrupt the screen. With a width of 14090 or less it doesn't happen.

Note Xorg doesn't actually crash, just displays garbage all over the screen and stops working.

I have looked in dmesg and the Xorg logs and couldn't find anything unusual.

I should note that matplotlib is using the Tk backend.

I realise that 14090 inches is an absurd value - I encountered this issue because this value was computed incorrectly from some other code. Cue several hours of confusion as I thought my graphics hardware was broken!

Apologies if this has nothing to do with matplotlib, but I'm unsure of how to debug this further.

**Code for reproduction**

```python
import matplotlib.pyplot as plt
plt.plot([1, 2, 3])
fig = plt.gcf()
crash = True
fig.set_size_inches(14090.5 if crash else 14090, 10)
plt.show()
```

**Actual outcome**

Xorg splats garbage all over the screen and becomes unresponsive.

**Expected outcome**

A window appears with the width equal to the width of the screen. This is what happens if I set crash = False in the above code.

**Matplotlib version**

  * Operating system: Debian 9.8
  * Matplotlib version: 2.0.0
  * Matplotlib backend: TkAgg
  * Python version: 3.5.3
  * Jupyter version (if applicable):
  * Other libraries: 

Python and matplotlib installed from Debian package repos.

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

Depending on the version of xserver creating a too big tk window
either corrupts xorg (!) or causes the process to exit.  This adds a
guard rail that will convert this into a python exception instead.

closes #17460

## PR Checklist

- [ ] Has Pytest style unit tests
- [k] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant

Not sure if it is worth adding a test for this as it is such an out-there condition.

### CODE REVIEW

File: lib/matplotlib/backends/_backend_tk.py [hunk 1]
```diff
@@ -431,6 +431,16 @@ def _get_toolbar(self):
         return toolbar
 
     def resize(self, width, height):
+        max_size = 1_409_023  # experimentally determined on xorg 1.20.8
+        if (width > max_size or height > max_size):
+            raise ValueError(
+                'You have requested to resize the '
+                f'Tk window to ({width}, {height}), one of which '
+                f'is bigger than {max_size}.  At larger sizes xorg will '
```
Comment (anntzer) [inline lib/matplotlib/backends/_backend_tk.py:439]: that message will make no sense on non-linux

Comment (tacaswell) [reply lib/matplotlib/backends/_backend_tk.py:439]: made the check linux only.

**Comment by jklymak** [comment]
Could we just have a warning for > 1000 inches?  Ie your figure is gigantic, and the backend may have difficulty?

**Comment by tacaswell** [comment]
In some combinations of tk and xserver we need to actually prevent the call to tk so we need more than a warning.  I'm hesitant to put a non externally motivated hard maximum size (8MB is big enough for anyone!).

**Comment by jklymak** [comment]
That’s why I suggested a warning.  In this case the user was accidentally setting the fig size to something crazy, and a warning would have helped debug the problem.  I don’t think we can or should add different guard rails for every system and backend combination

**Comment by tacaswell** [comment]
Can we do the warnings in a follow on?

File: lib/matplotlib/backends/_backend_tk.py [hunk 2]
```diff
@@ -431,6 +431,17 @@ def _get_toolbar(self):
...
 
     def resize(self, width, height):
+        max_size = 1_400_000  # the measured max on xorg 1.20.8 was 1_409_023
+
+        if (width > max_size or height > max_size) and sys.platform == 'linux':
+            raise ValueError(
+                'You have requested to resize the '
+                f'Tk window to ({width}, {height}), one of which '
+                f'is bigger than {max_size}.  At larger sizes xorg will '
+                'either exit with an error on never versions (~1.20) or '
```
Comment (timhoffm) [inline lib/matplotlib/backends/_backend_tk.py:441]:
```suggestion
                'either exit with an error on newer versions (~1.20) or '
```

**Pull request merged**